lib/Sema/CSSimplify.cpp

```diff
@@ -1141,20 +1141,41 @@ ConstraintSystem::matchTupleTypes(TupleType *tuple1, TupleType *tuple2,
     if (tuple1->getNumElements() != tuple2->getNumElements())
       return getTypeMatchFailure(locator);
 
+    // Determine whether this conversion is performed as part
+    // of a larger pattern matching operation.
+    bool inPatternMatchingContext = false;
+    {
+      SmallVector<LocatorPathElt, 4> path;
+      (void)locator.getLocatorParts(path);
+
+      inPatternMatchingContext =
+          llvm::any_of(path, [](const LocatorPathElt &elt) {
```

Comment:
Should we be more specific, by requiring the tail of the locator path to be some number of "function argument" locators before the `PatternMatch`? This looks like it will allow labels to be omitted even in other structural positions, e.g., making `[(Int, Int])` and `[(x: Int, y: Int)]` match.

Reply:
I can restrict that to have `pattern match` -> `function argument` at the end or just `pattern match`. I don't know whether there are other positions which this is important.

Reply:
Updated, does that make sense to you?